### ISSUE TITLE
tests/test_cli.py::test_encoding_* are broken with pytest-6: AttributeError: readonly attribute

### ISSUE THREAD

**Issue by mgorny** [andialbrecht/sqlparse#574]
It seems that pytest-6 doesn't permit touching `sys.stdout.encoding`:

```
$ tox -e py38
GLOB sdist-make: /tmp/sqlparse/setup.py
py38 create: /tmp/sqlparse/.tox/py38
py38 installdeps: pytest, pytest-cov, pytest-travis-fold
py38 inst: /tmp/sqlparse/.tox/.tmp/package/1/sqlparse-0.3.2.dev0.zip
py38 installed: attrs==20.1.0,coverage==5.2.1,iniconfig==1.0.1,more-itertools==8.4.0,packaging==20.4,pluggy==0.13.1,py==1.9.0,pyparsing==2.4.7,pytest==6.0.1,pytest-cov==2.10.1,pytest-travis-fold==1.3.0,six==1.15.0,sqlparse @ file:///tmp/sqlparse/.tox/.tmp/package/1/sqlparse-0.3.2.dev0.zip,toml==0.10.1
py38 run-test-pre: PYTHONHASHSEED='2004343036'
py38 run-test: commands[0] | sqlformat --version
0.3.2.dev0
py38 run-test: commands[1] | pytest --cov=sqlparse
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
cachedir: .tox/py38/.pytest_cache
rootdir: /tmp/sqlparse, configfile: setup.cfg
plugins: travis-fold-1.3.0, cov-2.10.1
collected 408 items                                                                                                                               

tests/test_cli.py ..........F.F.FF.                                                                                                         [  4%]
tests/test_format.py ............................................x.............x                                                            [ 18%]
tests/test_grouping.py .................................................................................................                    [ 42%]
tests/test_keywords.py ......                                                                                                               [ 43%]
tests/test_parse.py .....................................................................                                                   [ 60%]
tests/test_regressions.py ..............................................................................x..                                 [ 80%]
tests/test_split.py .........................                                                                                               [ 86%]
tests/test_tokenize.py ......................................................                                                               [100%]

==================================================================== FAILURES =====================================================================
____________________________________________________________ test_encoding_utf8_stdout ____________________________________________________________

filepath = <function filepath.<locals>.make_filepath at 0x7f66fa69ea60>, load_file = <function load_file.<locals>.make_load_file at 0x7f66fa69e9d0>
capfd = <_pytest.capture.CaptureFixture object at 0x7f66fa97ddc0>

    def test_encoding_utf8_stdout(filepath, load_file, capfd):
        path = filepath('encoding_utf8.sql')
        expected = load_file('encoding_utf8.sql', 'utf-8')
>       sys.stdout.encoding = 'utf-8'
E       AttributeError: readonly attribute

tests/test_cli.py:81: AttributeError
____________________________________________________________ test_encoding_gbk_stdout _____________________________________________________________

filepath = <function filepath.<locals>.make_filepath at 0x7f66fac873a0>, load_file = <function load_file.<locals>.make_load_file at 0x7f66fab4e3a0>
capfd = <_pytest.capture.CaptureFixture object at 0x7f66faa23d90>

    def test_encoding_gbk_stdout(filepath, load_file, capfd):
        path = filepath('encoding_gbk.sql')
        expected = load_file('encoding_gbk.sql', 'gbk')
>       sys.stdout.encoding = 'gbk'
E       AttributeError: readonly attribute

tests/test_cli.py:99: AttributeError
____________________________________________________________ test_encoding_stdin_utf8 _____________________________________________________________

filepath = <function filepath.<locals>.make_filepath at 0x7f66fac87430>, load_file = <function load_file.<locals>.make_load_file at 0x7f66fa69e820>
capfd = <_pytest.capture.CaptureFixture object at 0x7f66fa521d60>

    def test_encoding_stdin_utf8(filepath, load_file, capfd):
        path = filepath('encoding_utf8.sql')
        expected = load_file('encoding_utf8.sql', 'utf-8')
        old_stdin = sys.stdin
        with open(path, 'r') as f:
            sys.stdin = f
>           sys.stdout.encoding = 'utf-8'
E           AttributeError: readonly attribute

tests/test_cli.py:120: AttributeError
_____________________________________________________________ test_encoding_stdin_gbk _____________________________________________________________

filepath = <function filepath.<locals>.make_filepath at 0x7f66fa7b01f0>, load_file = <function load_file.<locals>.make_load_file at 0x7f66fa69eb80>
capfd = <_pytest.capture.CaptureFixture object at 0x7f66fa8ad910>

    def test_encoding_stdin_gbk(filepath, load_file, capfd):
        path = filepath('encoding_gbk.sql')
        expected = load_file('encoding_gbk.sql', 'gbk')
        old_stdin = sys.stdin
        with open(path, 'r') as stream:
            sys.stdin = stream
>           sys.stdout.encoding = 'gbk'
E           AttributeError: readonly attribute

tests/test_cli.py:133: AttributeError

----------- coverage: platform linux, python 3.8.5-final-0 -----------
Name                                    Stmts   Miss  Cover
-----------------------------------------------------------
sqlparse/__init__.py                       25      0   100%
sqlparse/cli.py                            66      6    91%
sqlparse/compat.py                         21      9    57%
sqlparse/engine/__init__.py                 4      0   100%
sqlparse/engine/filter_stack.py            25      0   100%
sqlparse/engine/grouping.py               262      0   100%
sqlparse/engine/statement_splitter.py      50      0   100%
sqlparse/exceptions.py                      2      0   100%
sqlparse/filters/__init__.py               13      0   100%
sqlparse/filters/aligned_indent.py         79      0   100%
sqlparse/filters/others.py                 77      1    99%
sqlparse/filters/output.py                 74      6    92%
sqlparse/filters/reindent.py              170      3    98%
sqlparse/filters/right_margin.py           28     17    39%
sqlparse/filters/tokens.py                 38      2    95%
sqlparse/formatter.py                     123      6    95%
sqlparse/keywords.py                       13      0   100%
sqlparse/lexer.py                          35      5    86%
sqlparse/sql.py                           339      7    98%
sqlparse/tokens.py                         37      0   100%
sqlparse/utils.py                          58      1    98%
-----------------------------------------------------------
TOTAL                                    1539     63    96%

============================================================= short test summary info =============================================================
FAILED tests/test_cli.py::test_encoding_utf8_stdout - AttributeError: readonly attribute
FAILED tests/test_cli.py::test_encoding_gbk_stdout - AttributeError: readonly attribute
FAILED tests/test_cli.py::test_encoding_stdin_utf8 - AttributeError: readonly attribute
FAILED tests/test_cli.py::test_encoding_stdin_gbk - AttributeError: readonly attribute
==================================================== 4 failed, 401 passed, 3 xfailed in 4.52s =====================================================
ERROR: InvocationError for command /tmp/sqlparse/.tox/py38/bin/pytest --cov=sqlparse (exited with code 1)
_____________________________________________________________________ summary _____________________________________________________________________
ERROR:   py38: commands failed
```

**Comment by andialbrecht** [comment]
Fixed in df25ba7484f2bacab00531e7e0638efc4fd5022f and df45180ec8b982810bd5837cf721da984e64faae

**Comment by mgorny** [comment]
Thank you!
